asu/util.py

```diff
@@ -115,7 +115,7 @@ def get_request_hash(req: dict) -> str:
             [
                 req.get("distro", ""),
                 req.get("version", ""),
-                req.get("version_code", ""),
+                req.get("version_code", "") or "",
```

Comment:
I think this isn't needed since .get defaults to "" already

Reply:
It is needed currently, because if you do
```
req["version_code"] = None       # The current default.
x = req.get("version_code", "")  # x is now None and 'join' crashes.
```
I'll rework, and see what happens if we just leave the default value out...

Reply:
Ok, seems I keep confusing typing's notion of optional and pydantic's optional...

```
field: str  = ""     # Pydantic optional because it has a default value
field: Optional[str] # Pydantic required because no default value
```